### ISSUE TITLE
chpldoc feature request: support `:noindexentry:`

### ISSUE THREAD

**Issue by jabraham17** [chapel-lang/chapel#23071]
Sphinx 3.2 added support for `:noindexentry:`, which is similar to `:noindex:` except that it allows cross-referencing using ``:ref:`...` ``.

We currently use `:noindex:` to prevent the internal modules from appearing in the index, but that prevents us from referencing them. So doing ``:mod:`owned <OwnedObject>` `` will fail since there is no entry for `OwnedObject`. But in this case, if we used `:noindexentry:`, we could use the cross-reference.

**Comment by mppf** [comment]
Would this work OK with the way we are embedding the internal module docs into the spec?

**Comment by jabraham17** [comment]
Yes, this works on every page in the spec and then queries a specific HTML structure which `chpldoc` generates. I've specifically used it on the owned/shared docs

**Comment by mppf** [comment]
A note from the sphinx docs

```


Changed in version 7.2:
    ...
    The directive option :noindexentry: was renamed to :no-index-entry:.
    ...
```
